### PR TITLE
feat: added the get_service instruction

### PR DESCRIPTION
**Pull request by h4ck3rk3y**
allows you to get a service object by name; useful if you lose a reference to a service or are using package imports

https://www.notion.so/kurtosistech/get_service-and-port-url-b81ad4110c294ddc8851a7ec0ffe2932

### CODE REVIEW

File: core/server/api_container/server/startosis_engine/interpretation_time_value_store/service_value_repository.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+		// check for existence
+		if serviceSerializedValue == nil {
+			return stacktrace.NewError("Service '%v' doesn't exist in the repository", name)
+		}
+
+		isEmptyValue := len(serviceSerializedValue) == len(emptyValue)
+
+		serviceSerializedValueStr := string(serviceSerializedValue)
+
+		// this will the case if the key was saved with an empty value
```
Comment (tedim52) [inline core/server/api_container/server/startosis_engine/interpretation_time_value_store/service_value_repository.go]: `this will catch the case`?

File: core/server/api_container/server/startosis_engine/interpretation_time_value_store/service_value_repository.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+	repository := &serviceInterpretationValueRepository{
+		enclaveDb:          enclaveDb,
+		starlarkValueSerde: starlarkValueSerde,
+	}
+
+	return repository, nil
+}
+
+func (repository *serviceInterpretationValueRepository) AddService(name service.ServiceName, service *kurtosis_types.Service) error {
```
Comment (tedim52) [inline core/server/api_container/server/startosis_engine/interpretation_time_value_store/service_value_repository.go]: nit: suggest renaming `AddService` to `PutService` bc `AddService` means something else in the startosis engine (add in service network) so it's an overloaded concept

File: core/server/api_container/server/startosis_engine/interpretation_time_value_store/service_value_repository.go (hunk 1 above)
Comment (tedim52) [reply core/server/api_container/server/startosis_engine/interpretation_time_value_store/service_value_repository.go]: What cases would it be valid to store `service` with an empty value? If it's not valid, maybe we can prevent it from being stored as an empty value in `AddService`?

File: core/server/api_container/server/startosis_engine/interpretation_time_value_store/service_value_repository_test.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	err := repository.AddService(serviceName, expectedService)
+	require.Nil(t, err)
+
+	actualService, err := repository.GetService(serviceName)
+	require.Nil(t, err)
+	require.Equal(t, expectedService.AttrNames(), actualService.AttrNames())
+	require.Equal(t, expectedService.String(), actualService.String())
+}
+
+func getRecipeResultRepositoryForTest(t *testing.T) *serviceInterpretationValueRepository {
```
Comment (tedim52) [inline core/server/api_container/server/startosis_engine/interpretation_time_value_store/service_value_repository_test.go]: is `Recipe` supposed to indicate a template of some sort?

File: core/server/api_container/server/startosis_engine/interpretation_time_value_store/service_value_repository_test.go [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+const (
+	starlarkThreadName     = "thread-for-db-test"
+	serviceName            = service.ServiceName("datastore-1")
+	serviceNameStarlarkStr = starlark.String(serviceName)
+	hostName               = serviceNameStarlarkStr
+	ipAddress              = starlark.String("172.23.34.44")
+)
+
+func TestAddAndGetTest(t *testing.T) {
```
Comment (tedim52) [inline core/server/api_container/server/startosis_engine/interpretation_time_value_store/service_value_repository_test.go]: maybe add a test for the service doesn't exist case and the case where the service was stored as an empty value to illustrate/test those behaviors?

File: core/server/api_container/server/startosis_engine/interpretation_time_value_store/service_value_repository.go (hunk 1 above)
Comment (h4ck3rk3y) [reply core/server/api_container/server/startosis_engine/interpretation_time_value_store/service_value_repository.go]: Yeah, i had got this from recipe but now i am throwing an error!

**Review by tedim52 — APPROVED** [review]
